distributed/worker.py

```diff
@@ -1284,13 +1308,14 @@ async def close(
                 with suppress(TimeoutError):
                     await self.batched_stream.close(timedelta(seconds=timeout))
 
-            self.actor_executor._work_queue.queue.clear()
-            if isinstance(self.executor, ThreadPoolExecutor):
-                self.executor._work_queue.queue.clear()
-                self.executor.shutdown(wait=executor_wait, timeout=timeout)
-            else:
-                self.executor.shutdown(wait=False)
-            self.actor_executor.shutdown(wait=executor_wait, timeout=timeout)
+            for executor in self.executors.values():
+                if executor is utils._offload_executor:
+                    continue  # Never shutdown the offload executor
+                if isinstance(executor, ThreadPoolExecutor):
+                    executor._work_queue.queue.clear()
```

Comment:
I'm curious why we're doing this step manually (I realize this predates this PR -- this isn't meant to be a blocker for merging the changes here)

Reply:
It is because even with `shutdown(wait=False)`, the `ThreadPoolExecutor` will still run pending tasks. By calling `executor._work_queue.queue.clear()`, we cancel them before shutting down.

Reply:
Ah, I see. Thanks for clarifying! FWIW it looks like a [`cancel_futures=` keyword](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown) was added to `Executor.shutdown` in Python 3.9. Just pointing this out as something we can use in the future once 3.9 is our minimum Python version